mcs/class/System/ReferenceSources/Win32Exception.cs

```diff
@@ -1738,6 +1753,1722 @@ internal static string GetErrorMessage (int error)
 			default:
 				return string.Format ("mono-io-layer-error ({0})", error);
 			}
+#endif // UNITY
+	}
+
+#if UNITY
+		static void InitializeErrorMessages()
+		{
+			if (s_ErrorMessagesInitialized)
+				return;
+
+			s_ErrorMessagesInitialized = true;
```

Comment:
Nit, this is racy. Ideally we'd take a lock but at very least we should not set `s_ErrorMessagesInitialized ` to `true` until the end of this initialization.

Although, if we race in that case we'll get exception since you can't add duplicate keys. So, lock?

Reply:
Or just put this in static constructor and let runtime handle for you.

Reply:
I've fixed this issue.